src/CodeGen_Xtensa.cpp

```diff
@@ -0,0 +1,1593 @@
+            } else {
+                print_assignment(vector_type, "dense_ramp<" + print_type(vector_type) + ">(" + id_base + ")");
+            }
+        }
+    } else {
+        if (is_native_xtensa_vector<int32_t>(op->type, target)) {
+            print_assignment(vector_type, "/* ramp */ int32x" + std::to_string(int32_lanes) + "_t(" + id_base + ") + IVP_PACKLN_2X64W(IVP_SEQN_2X32() * int32x" + std::to_string(int32_lanes) + "_t(" + id_stride + "))");
+        } else if ((op->type.lanes() == 32 || op->type.lanes() == 64 || op->type.lanes() == 128) && op->type.is_int_or_uint() && op->type.bits() == 32) {
+            print_assignment(vector_type, "ramp<" + print_type(vector_type) + ">(" + id_base + ", " + id_stride + ")");
+        } else {
+            print_assignment(vector_type, print_type(vector_type) + "_ops::ramp(" + id_base + ", " + id_stride + ")");
+        }
+    }
+}
+
+void CodeGen_Xtensa::visit(const Broadcast *op) {
+    Type vector_type = op->type.with_lanes(op->lanes);
+    string rhs;
+    if (op->type.is_int() && ((op->type.bits() == 24) || (op->type.bits() == 48)) && is_const(op->value)) {
+        // Assigning a constant to wide vector is tricky.
+        if (is_const_zero(op->value)) {
+            if (op->type.bits() == 24) {
+                rhs = "IVP_ZERO2NX24()";
+            } else if (op->type.bits() == 48) {
+                rhs = "IVP_ZERONX48()";
+            }
+        } else {
+            rhs = std::to_string(op->value.as<IntImm>()->value);
+        }
+    } else if (op->type.is_int_or_uint() && op->type.bits() == 8 && ((op->type.lanes() == 4) || (op->type.lanes() == 8))) {
+        string id_value = print_expr(op->value);
+        rhs = "broadcast<" + print_type(op->type) + ", " + print_type(op->value.type()) + ">(" + id_value + ")";
+    } else {
+        string id_value = print_expr(op->value);
+
+        if (is_native_vector_type(op->type, target)) {
+            // TODO(vsknk): why it this extra cast to scalar is needed?
+            rhs = print_type(vector_type) + "((" + print_type(op->type.with_lanes(1)) + ")" + id_value + ")";
+        } else if (op->lanes > 1) {
+            if (op->type.is_bool()) {
+                // TODO(vksnk): figure out how to broadcast bool.
+                if (op->type.lanes() == 16) {
+                    rhs = id_value + "? (int32x16_t(1) == int32x16_t(1)) : (int32x16_t(1) == int32x16_t(0))";
+                } else if (op->type.lanes() == 32) {
+                    rhs = id_value + "? (int16x32_t(1) == int16x32_t(1)) : (int16x32_t(1) == int16x32_t(0))";
+                } else if (op->type.lanes() == 64) {
+                    rhs = id_value + "? (int8x64_t(1) == int8x64_t(1)) : (int8x64_t(1) == int8x64_t(0))";
+                }
+            } else {
+                rhs = id_value;
+            }
+        } else {
+            rhs = id_value;
+        }
+    }
+
+    print_assignment(vector_type, rhs);
+}
+
+void CodeGen_Xtensa::visit(const LE *op) {
```

Comment:
This method, the `LT` visitor, the `GE` visitor, and the `GT` visitor all seem to share the same logic, and could probably benefit from a single method that is parameterized by an op string. I think the `Min` and `Max` visitor might be the same exact logic as well, but that's just me squinting at it.

Reply:
Probably the `EQ` visitor too.